src/ctapipe/calib/camera/extractor.py

```diff
@@ -0,0 +1,256 @@
+"""
+Extraction algorithms to compute the statistics from a chunk of images
+"""
+
+__all__ = [
+    "StatisticsExtractor",
+    "PlainExtractor",
+    "SigmaClippingExtractor",
+]
+
+from abc import abstractmethod
+
+import numpy as np
+from astropy.stats import sigma_clipped_stats
+
+from ctapipe.containers import StatisticsContainer
+from ctapipe.core import TelescopeComponent
+from ctapipe.core.traits import (
+    CaselessStrEnum,
+    Int,
+    List,
+)
+
+
+class StatisticsExtractor(TelescopeComponent):
+    """
+    Base component to handle the extraction of the statistics from a dl1 table
+    containing charges, peak times and/or charge variances (images).
+    """
+
+    chunk_size = Int(
+        2500,
+        help="Size of the chunk used for the calculation of the statistical values",
+    ).tag(config=True)
+
+    def __call__(
+        self,
+        dl1_table,
+        masked_pixels_of_sample=None,
+        chunk_shift=None,
+        col_name="image",
+    ) -> list:
+        """
+        Prepare the extraction chunks and call the relevant function of the particular extractor
+        to extract the statistical values.
+
+        Parameters
+        ----------
+        dl1_table : astropy.table.Table
+            DL1 table with images of shape (n_images, n_channels, n_pix)
+            and timestamps of shape (n_images, ) stored in an astropy Table
+        masked_pixels_of_sample : ndarray
+            boolean array of masked pixels of shape (n_pix, ) that are not available for processing
+        chunk_shift : int
+            number of samples to shift the extraction chunk
+        col_name : string
+            column name in the DL1 table
+
+        Returns
+        -------
+        List StatisticsContainer:
+            List of extracted statistics and extraction chunks
+        """
+
+        # Check if the length of the dl1 table is greater or equal than the size of the chunk.
+        if len(dl1_table[col_name]) < self.chunk_size:
+            raise ValueError(
+                f"The length of the DL1 table ({len(dl1_table[col_name])}) must be greater or equal than the size of the chunk ({self.chunk_size})."
+            )
+
+        # Function to split table data into appropriated chunks
+        def _get_chunks(dl1_table_data, chunk_shift):
+            if chunk_shift is None:
```

Comment:
I think this could just use the same code if you use `chunk_shift=0` as default instead of `chunk_shift=None`:

Reply:
Actually, `chunk_shift=self.chunk_size` once `chunk_shift is None`. This is the first pass, where we do not allow overlapping chunks except in the border case at the end of the sequence. The last extraction chunk of the first pass will be performed with the last available elements of the sequence in order to fill a complete chunk. For the second pass, we detect faulty chunks and perform a detail scan of the region of trouble with a sliced dl1 table (around that region). Therefore, we utilize the `chunk_shift` argument. If the detail scan is performed we do not want to include the border case as before in order to avoid repetitions of the last complete chunk every time the chunk is shifted.

Reply:
That's not what I meant. I meant that the `if / else` block here with very similar code could be avoided if "no shift" would be `chunk_shift=0` instead of `chunk_shift=None`

Reply:
We bypassed the `if/else` bock by `yield`ing non-overflow chunks. If requested the last complete chunk will be `yield` as well. Can you please check if the function is clearer now? Thanks!